spring-cloud-dataflow-server-core/src/main/resources/schema-hsqldb.sql

```diff
@@ -1,6 +1,15 @@
 
-CREATE TABLE TASK_DEFINITION  (
+CREATE TABLE TASK_DEFINITIONS  (
 	DEFINITION_NAME VARCHAR(255) NOT NULL PRIMARY KEY,
 	DEFINITION CLOB DEFAULT NULL
 );
 
+CREATE TABLE STREAM_DEFINITIONS  (
+	DEFINITION_NAME VARCHAR(255) NOT NULL PRIMARY KEY,
+	DEFINITION CLOB DEFAULT NULL
+);
+
+CREATE TABLE DEPLOYMENT_IDS  (
+	DEPLOYMENT_KEY VARCHAR(255) NOT NULL PRIMARY KEY,
+	DEPLOYMENT_ID VARCHAR(255) NOT NULL
+);
```

Comment:
Yeah, I see. I think we can check if the deploymentId exists before saving. That way we can be sure of it.